pkg/codegen/dotnet/gen_program_expressions.go

```diff
@@ -400,24 +447,45 @@ func (g *generator) GenFunctionCallExpression(w io.Writer, expr *model.FunctionC
 		// Assuming the existence of the following helper method located earlier in the preamble
 		g.Fgenf(w, "ComputeFileBase64Sha256(%v)", expr.Args[0])
 	case pcl.Invoke:
-		_, name := g.functionName(expr.Args[0])
-
-		isOut, outArgs, outArgsTy := pcl.RecognizeOutputVersionedInvoke(expr)
-		if isOut {
-			g.visitToMarkTypesUsedInFunctionOutputVersionInputs(outArgs)
-			g.Fprintf(w, "%s.Invoke(", name)
-			typeName := g.argumentTypeNameWithSuffix(expr, outArgsTy, "InvokeArgs")
-			g.genObjectConsExpressionWithTypeName(w, outArgs, typeName)
+		_, fullFunctionName := g.functionName(expr.Args[0])
+		functionParts := strings.Split(fullFunctionName, ".")
+		functionName := functionParts[len(functionParts)-1]
+		if g.insideAwait {
+			g.Fprintf(w, "%s.InvokeAsync(", fullFunctionName)
 		} else {
-			g.Fprintf(w, "%s.InvokeAsync(", name)
-			if len(expr.Args) >= 2 {
-				g.Fgenf(w, "%.v", expr.Args[1])
-			}
+			g.Fprintf(w, "%s.Invoke(", fullFunctionName)
 		}
 
-		if len(expr.Args) == 3 {
-			g.Fgenf(w, ", %.v", expr.Args[2])
+		innerFunc, isFunc := expr.Args[1].(*model.FunctionCallExpression)
+		if isFunc && innerFunc.Name == pcl.IntrinsicConvert {
```

Comment:
Is this the correct behavior for `isFunc && innerFunc.Name != pcl.IntrensicConcert`?

Reply:
I believe so, this logic only special-cases that specific pattern and the default case is what it was before